custom_components/samsungtv_tizen/media_player.py

```diff
@@ -431,6 +488,9 @@ def media_title(self):
     @property
     def state(self):
         """Return the state of the device."""
+        if self._power_off_in_progress():
```

Comment:
But `self._power_off_in_progress()` is `True` even if I use the turn off method and the TV did not actually turn off. While this logic presumes that it's off just because I tried to turn it off, this is false data as nothing confirmed the TV as actually being off..

Normally I get the actual state of the TV in about 10 seconds (with the ping update method), and you set `POWER_OFF_DELAY` to 20 seconds. So for 10 seconds it will lie to me that it's off (when it doesn't really know), and then for 10 more seconds (even after it knows that the TV is actually still on) it will keep telling me it's off until the timer clears.

Reply:
I'm optimistic and I assume that the TV answer to the command. In this way again I have a immediate feed-back of the command sent (Power Off), against having TV physically Off but control still active.
Related to the timing, update function runs every 15 sec (const SCAN_INTERVAL defined at the begining - default is 30 sec.). If the fist update run after 5 sec. that we send the power off, probably update still return ON and we have to wait additional 15 sec. So I consider 20 sec. an acceptable compromise. As alternative option we could reduce SCAN_INTERVAL, but I don't think is a good option going too low with this.

Reply:
Ok, leave it at 20 seconds but pls add a comment above it stating that it is based on an asumption and for 20 seconds the state will be off regardless of the actual state of the TV. Just in case this leads to unforseable issues in the future (which it might).

Reply:
done